src/main.go

```diff
@@ -0,0 +1,320 @@
+
+	wellArchitected.logger.Logf("INFO", "Finished successfully")
+
+	return nil
+}
+
+func (wa *WellArchitected) collectData() error {
+	workloadSummaries, err := wa.getWorkloadSummaries()
+
+	if err != nil {
+		return err
+	}
+
+	for _, workloadSummary := range workloadSummaries {
+		workloadID := workloadSummary.WorkloadId
+		workload, err := wa.getWorkload(workloadID)
+
+		if err != nil {
+			return err
+		}
+
+		for _, lens := range workload.Lenses {
+			lensReview, err := wa.getLensReview(workloadID, &lens)
+
+			if err != nil {
+				return err
+			}
+
+			pillarReviewSummaries := lensReview.LensReview.PillarReviewSummaries
+
+			for _, pillarReviewSummary := range pillarReviewSummaries {
+				err := wa.parseLensReview(lensReview, &pillarReviewSummary)
+
+				if err != nil {
+					return err
+				}
+			}
+
+			lensReviewImprovements, err := wa.getLensReviewImprovements(workloadID, &lens)
+
+			if err != nil {
+				return err
+			}
+
+			improvementSummaries := lensReviewImprovements.ImprovementSummaries
+
+			for _, improvementSummary := range improvementSummaries {
+				err := wa.parseLensReviewImprovements(lensReviewImprovements, &improvementSummary)
+
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (wa *WellArchitected) getWorkloadSummaries() ([]types.WorkloadSummary, error) {
+	listWorkloadsInput := &wellarchitected.ListWorkloadsInput{}
+	listWorkloadsOutput, err := wa.client.ListWorkloads(wa.ctx, listWorkloadsInput)
+
+	if err != nil {
+		return nil, fmt.Errorf("did not get workload summaries: %v", err)
+	}
+
+	return listWorkloadsOutput.WorkloadSummaries, nil
+}
+
+func (wa *WellArchitected) getWorkload(workloadID *string) (*types.Workload, error) {
+	workloadInput := &wellarchitected.GetWorkloadInput{WorkloadId: workloadID}
+	workloadOutput, err := wa.client.GetWorkload(wa.ctx, workloadInput)
+
+	if err != nil {
+		return nil, fmt.Errorf("did not get workload with id %s: %v", *workloadID, err)
+	}
+
+	workloadJSON, err := json.Marshal(workloadOutput.Workload)
+
+	if err != nil {
+		return nil, fmt.Errorf("error marshaling workload with id %s: %v", *workloadID, err)
+	}
+
+	workloadJSON, err = addTypeToData(workloadJSON)
+
+	if err != nil {
+		return nil, err
+	}
+
+	wa.data = append(wa.data, workloadJSON)
+
+	return workloadOutput.Workload, nil
+}
+
+func (wa *WellArchitected) getLensReview(workloadID *string, lensAlias *string) (*wellarchitected.GetLensReviewOutput, error) {
+	lensReviewInput := &wellarchitected.GetLensReviewInput{
+		WorkloadId: workloadID,
+		LensAlias:  lensAlias,
+	}
+
+	lensReviewOutput, err := wa.client.GetLensReview(wa.ctx, lensReviewInput)
+
+	if err != nil {
+		return nil, fmt.Errorf("did not get lens review with workload id %s and lens alias %s: %v", *workloadID, *lensAlias, err)
+	}
+
+	return lensReviewOutput, nil
+}
+
+func (wa *WellArchitected) parseLensReview(lensReview *wellarchitected.GetLensReviewOutput, pillarReviewSummary *types.PillarReviewSummary) error {
```

Comment:
Maybe rename the func so that it will indicate that it not only parses the data, but also adds it to wa?